### PR TITLE
lib: os: cbprintf: Fixes for sparc and size calculation of void * argument

### PR DESCRIPTION
**Pull request by nordic-krch**
Similarly to runtime packaging, sparc architecture needed special treatment for unaligned storing of 64/128 bit arguments.
Additionally, there was an error in calculation of `void *` size in static packaging and description of `Z_CBPRINTF_ARG_SIZE` which relaying on older concept for cbprintf packaging.

### CODE REVIEW

**Review by npitre — APPROVED** [review]

**Review by pabigot — APPROVED** [review]
I don't see anything obviously problematic about this, but SPARC is indeed strange, and I'm not sure whether register windows have any relevance to this.  @martin-aberg as SPARC maintainer should be a required reviewer for this PR, even if my ACK brings this up to "mergeable".

**Comment by nordic-krch** [comment]
@npitre in cbprintf you have used defines like `__xtensa__`, `__i386__` to distinguish between platforms. I am not sure if they are C or GCC specific. Zephyr in general is using zephyr specific defines from Kconfig (e.g. `CONFIG_XTENSA`). Are you ok if i change them to those zephyr specific equivalents?

**Comment by npitre** [comment]
I don't think that is a good idea. I used compiler specific defines 
because this va_list game is somewhat compiler specific too. This 
appears to be the case at least for Xtensa with no official ABI 
documentation I could find. In the end there is nothing actually Zephyr 
specific here either.

I'd prefer to wait until some other compiler without the gcc/clang 
defines shows up and potentially break the test due to unsuitable 
default settings so that we can have a closer look right away.

File: include/sys/cbprintf_internal.h [hunk 1]
```diff
@@ -120,6 +120,26 @@ extern "C" {
  *
  * @param arg Argument.
  */
+#ifdef __sparc__
+static inline void cbprintf_wcpy(int *dst, int *src, uint32_t len)
+{
+	for (int i = 0; i < len; i++) {
+		dst[i] = src[i];
+	}
+}
```
Comment (martin-aberg) [inline include/sys/cbprintf_internal.h:129]: Yes, all `<stdarg.h>` variable list arguments are 32-bit aligned on the stack. So it should be fine.

The SPARC V8 ABI specification, page 3-36, describes how variable list arguments are stored, together with an example.
https://www.gaisler.com/doc/sparc-abi.pdf#page=50


I would recommend that the first two sentences of the commit message are rephrased, in line with:

```
The SPARC V8 ABI guarantees that the arguments of a variable argument
list function are stored on the stack at addresses which are 32-bit
aligned. It means that variables of type unit64_t and double may not
be properly aligned on the stack.

The compiler is aware of the ABI and takes care of this. However,
as we are directly accessing the variable argument list here, we need
to take the alignment into consideration and copy 64-bit arguments
as 32-bit words.
```

Comment (nordic-krch) [reply include/sys/cbprintf_internal.h:129]: I've added this as a comment in the code when defining a special treatment in #33473

**Pull request merged**